andy/src/main/java/nl/tudelft/cse1110/andy/execution/step/GetRunConfigurationStep.java

```diff
@@ -25,16 +25,43 @@ public void execute(Context ctx, ResultBuilder result) {
             Class<?> runConfigurationClass = Class.forName(getConfigurationClass(ctx.getNewClassNames()), false, currentClassLoader);
             RunConfiguration runConfiguration = (RunConfiguration) runConfigurationClass.getDeclaredConstructor().newInstance();
 
+            if(!isValidRunConfiguration(runConfiguration)) {
+                // load default configuration
+                ctx.setRunConfiguration(new DefaultRunConfiguration(allClassesButTestingAndConfigOnes(ctx.getNewClassNames())));
+                throw new RuntimeException("The run configuration has invalid weights. Please check the file.");
+            }
+
             ctx.setRunConfiguration(runConfiguration);
         } catch (NoSuchElementException ex) {
-            // There's no configuration set. We put a default one!
-            RunConfiguration runConfiguration = new DefaultRunConfiguration(allClassesButTestingAndConfigOnes(ctx.getNewClassNames()));
-            ctx.setRunConfiguration(runConfiguration);
+            // load default configuration
+            ctx.setRunConfiguration(new DefaultRunConfiguration(allClassesButTestingAndConfigOnes(ctx.getNewClassNames())));
         } catch (Exception ex) {
             result.genericFailure(this, ex);
         }
     }
 
+    /**
+     * Checks whether the config is valid.
+     * - Sum of weights must be 1.0
+     * - If Jacoco / Pitest are skipped, the weight should be 0
+     * @param config The run configuration of the problem
+     * @return validity of the config weights represented as a boolean
+     */
+    public static boolean isValidRunConfiguration(RunConfiguration config) {
```

Comment:
This method would look better inside the `RunConfiguration` class itself. Look at it: 99% of it is interacting with data from that class, this is a sign that this method should be there.